Log graphics API that is in use by the Qt Quick

hebasto opened the pull request:
Some log excerpts with this PR:

- Linux Mint, system Qt 5.12.8
```
$ ./src/qt/bitcoin-qt -printtoconsole
Bitcoin Core version v22.99.0-76426d3948b2 (release build)
Qt 5.12.8 (dynamic), plugin=xcb (dynamic)
No static plugins.
QQuickStyle: Default
System: Linux Mint 20.2, x86_64-little_endian-lp64
Screen: HDMI-1 2560x1440, pixel ratio=1.0
Screen: DP-1 1920x1080, pixel ratio=1.0
...
GUI: Graphics API in use: "OpenGL ES 2.0 or higher"
...
```

- Ubuntu Hirsute, system Qt 5.15.2
```
$ QSG_RHI=1 ./src/qt/bitcoin-qt -printtoconsole
GUI: Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
Bitcoin Core version v22.99.0-76426d3948b2 (release build)
Qt 5.15.2 (dynamic), plugin=xcb (dynamic)
No static plugins.
QQuickStyle: Default
System: Ubuntu 21.04, x86_64-little_endian-lp64
Screen: XWAYLAND0 2560x1440, pixel ratio=1.0
...
GUI: Graphics API in use: "OpenGL ES 2.0 or higher via a graphics abstraction layer"
...
```

promag commented on src/qml/bitcoin.cpp:
```diff
@@ -146,5 +149,7 @@ int QmlGuiMain(int argc, char* argv[])
         return EXIT_FAILURE;
     }
 
+    qInfo() << "Graphics API in use:" << QmlUtil::GraphicsApi(qobject_cast<QQuickWindow*>(engine.rootObjects().first()));
```
Since we expecting a window I'd also do:

```cpp
auto window = qobject_cast<QQuickWindow*>(engine.rootObjects().first());
if (!window) return EXIT_FAILURE;

qInfo() << "Graphics API in use:" << QmlUtil::GraphicsApi(window);
```

hebasto replied:
Thanks! [Updated](https://github.com/bitcoin-core/gui-qml/pull/32#issuecomment-908914269).

promag commented on src/qml/util.h:
```diff
@@ -0,0 +1,27 @@
+namespace QmlUtil {
+/**
+     * Returns a human-readable description of the graphics API
+     * that is in use by the Qt Quick scenegraph.
```
nit, ... scenegraph renderer.

hebasto replied:
Thanks! [Updated](https://github.com/bitcoin-core/gui-qml/pull/32#issuecomment-908914269).

promag reviewed (commented):
Concept ACK.

jarolrod commented on src/qml/util.cpp:
```diff
@@ -0,0 +1,34 @@
+    case QSGRendererInterface::OpenGL: return "OpenGL ES 2.0 or higher";
+    case QSGRendererInterface::Direct3D12: return "Direct3D 12";
+    case QSGRendererInterface::OpenVG: return "OpenVG via EGL";
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 14, 0))
```
There is the [isApiRhiBased function](https://doc-snapshots.qt.io/qt6-dev/qsgrendererinterface.html#isApiRhiBased). It was introduced in Qt 5.14. 

This check is then appropriate and allows for a simplified control flow.

promag replied:
@jarolrod still needs `#if` as the cases below don't exist in <5.14.

jarolrod replied:
yep, i meant that the way it is done here simplifies the control flow compared to if we used the function.

jarolrod commented on src/qml/util.cpp:
```diff
@@ -0,0 +1,34 @@
+QString GraphicsApi(QQuickWindow* window)
+{
+    switch (window->rendererInterface()->graphicsApi()) {
+    case QSGRendererInterface::Unknown: return "Unknown";
```
I've spent hours and hours digging through Qt code to figure out if there is a better way than manually maintaining these strings. Sadly, there is not. For now, this allows us to conveniently debug. 

Note: Let's revisit this at some point in the future

jarolrod commented:
Concept ACK

hebasto commented:
Updated c79be815a479e43c4d2fe3eac8e269806aefaa4b -> 76426d3948b28e85bb767ddcc645b5344cc7aaaa ([pr32.01](https://github.com/hebasto/gui-qml/commits/pr32.01) -> [pr32.02](https://github.com/hebasto/gui-qml/commits/pr32.02)):

- rebased
- addressed @promag's comments

hebasto commented:
FWIW, setting `QSG_INFO=1` also adds more debugging info.

hebasto commented:
The PR description has been updated.

promag reviewed (commented):
Code review ACK 76426d3948b28e85bb767ddcc645b5344cc7aaaa.
